packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -170,22 +224,37 @@ export class PanelPlugin<TOptions = any> extends GrafanaPlugin<PanelPluginMeta>
    *
    * @public
    **/
-  setPanelOptions(builder: (builder: PanelOptionsEditorBuilder) => void) {
+  setPanelOptions(builder: (builder: PanelOptionsEditorBuilder<TOptions>) => void) {
     // builder is applied lazily when options UI is created
     this.registerOptionEditors = builder;
     return this;
   }
 
   /**
-   * Enables configuration of panel's default field config
+   * Allows specyfing which common field config options panel should use
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * // when plugin should use only specific common options
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *  .useCommonFieldConfig([StandardFieldConfigProperties.Min, StandardFieldConfigProperties.Max, StandardFieldConfigProperties.Links]);
+   *
+   * // when plugin should use all common options
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *  .useCommonFieldConfig();
```

Comment:
Doing nothing should have all the common fields used?  If you want to hide them could it be:
```
.useCommonFieldConfig([])
```
or maybe `showCommonFieldConfig`?

Reply:
yea, if you want to hide them all I think .useCommonFieldConfig([]) is best, so only need one function to control this

Reply:
@torkelo maybe by default none should be shown unless u use `useCommonFieldConfig`? I think this is most intuitive approach

Reply:
I do not know what default is best (get all standard fields) or optin using useStandardFieldConfig()  with no args.,